subiquity/models/mirror.py

```diff
@@ -13,35 +13,59 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import copy
 import logging
-import platform
 from urllib import parse
 
+from curtin.commands.apt_config import (
+    get_arch_mirrorconfig,
+    get_mirror,
+    PRIMARY_ARCHES,
+    )
+from curtin.util import get_architecture
+
 log = logging.getLogger('subiquitycore.models.mirror')
 
-# correct default mirror for most arches
-DEFAULT_MIRROR = 'http://ports.ubuntu.com/ubuntu-ports'
-# apart from the two snowflakes
-if platform.machine() in ['i686', 'x86_64']:
-    DEFAULT_MIRROR = 'http://archive.ubuntu.com/ubuntu'
+
+DEFAULT = {
+    "preserve_sources_list": False,
+    "primary": [
+        {
+            "arches": PRIMARY_ARCHES,
+            "uri": "http://archive.ubuntu.com/ubuntu",
+        },
+        {
+            "arches": ["default"],
+            "uri": "http://ports.ubuntu.com/ubuntu-ports",
+        },
+        ],
+}
+
+ARCHITECTURE = get_architecture()
```

Comment:
Doesn't this create a side-effect of executing dpkg if something imports this module?

Reply:
AFAICT, nothing outside of this module uses DEFAULT_MIRROR or ARCHITECTURE and the API being used are the method son the MirrorModel itself, could we move these values to the class itself? then they can be properties that are cached after initialization in __init__?

Reply:
Yeah, that makes more sense, thanks.